src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -195,10 +195,19 @@ private ManagedWebSocket(Stream stream, bool isServer, string subprotocol, TimeS
                 }
             }, this);
 
-            // Now that we're opened, initiate the keep alive timer to send periodic pings
+            // Now that we're opened, initiate the keep alive timer to send periodic pings.
+            // We use a weak reference from the timer to the web socket to avoid a cycle
+            // that could keep the web socket rooted in erroneous cases.
             if (keepAliveInterval > TimeSpan.Zero)
```

Comment:
Ideally this timer wouldn't start on the client side until ConnectAsync is called.